### PR TITLE
New Sample: Query features with arcade expression

### PR DESCRIPTION
**Pull request by Gela**
# Description

## Type of change

- [ ] Bug fix
- [x] New sample implementation
- [ ] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [x] Windows
- [ ] Android
- [ ] Linux
- [ ] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [ ] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        if (results.empty())
+        {
+          return;
+        }
+        QList<GeoElement*> element_list = results.first()->geoElements();
+
+        if (element_list.empty())
+        {
+          return;
+        }
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp:112]:
```suggestion
        if (element_list.empty())
          return;
```

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      Point mapPoint(m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y()));
+      m_mapView->calloutData()->setLocation(mapPoint);
+      m_mapView->calloutData()->setVisible(true);
+
+      connect(m_mapView, &MapQuickView::identifyLayersCompleted, this, [this, mapPoint](QUuid, const QList<IdentifyLayerResult*>& results)
+      {
+        if (results.empty())
+        {
+          return;
+        }
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp:106]:
```suggestion
        if (results.empty())
          return;
```

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if (m_mapView->calloutData()->isVisible())
+      m_mapView->calloutData()->setVisible(false);
+    else
+    {
+      // Set callout position
+      Point mapPoint(m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y()));
+      m_mapView->calloutData()->setLocation(mapPoint);
+      m_mapView->calloutData()->setVisible(true);
+
+      connect(m_mapView, &MapQuickView::identifyLayersCompleted, this, [this, mapPoint](QUuid, const QList<IdentifyLayerResult*>& results)
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp:101]: This will create a new slot every time `mouseClicked` fires

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          return;
+        }
+        QList<GeoElement*> element_list = results.first()->geoElements();
+
+        if (element_list.empty())
+        {
+          return;
+        }
+        GeoElement* element = element_list.at(0);
+        Esri::ArcGISRuntime::ArcGISFeature* identifiedFeature = dynamic_cast<ArcGISFeature*>(element);
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp:114]:
```suggestion
        ArcGISFeature* identifiedFeature = dynamic_cast<ArcGISFeature*>(element);
```

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  m_mapView->calloutData()->setVisible(false);
+  m_mapView->calloutData()->setTitle("RPD Beats");
+
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& mouseEvent){
+    if (m_mapView->calloutData()->isVisible())
+      m_mapView->calloutData()->setVisible(false);
+    else
+    {
+      // Set callout position
+      Point mapPoint(m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y()));
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.cpp:97]:
```suggestion
      const Point mapPoint(m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y()));
```

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+## Use case
+Arcade is a portable, lightweight, and secure expression language used to create custom content in ArcGIS applications. Like other expression languages, it can perform mathematical calculations, manipulate text, and evaluate logical statements. It also supports multi-statement expressions, variables, and flow control statements. What makes Arcade particularly unique when compared to other expression and scripting languages is its inclusion of feature and geometry data types. This sample uses an arcade expression to query the number of crimes in a neighborhood in the last 60 days.
+
+## How to use the sample
+Click on any neighborhood to see the number of crimes in the last 60 days in a callout.
+
+## How it works
+1. Create a `PortalItem` using the URL and ID.
+2. Create a `Map` using the portal item.
+3. Set the visibility of all the layers to false, except for the layer named "RPD Beats  - City_Beats_Border_1128-4500".
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:16]:
```suggestion
3. Set the visibility of all the layers to false, except for the layer named "RPD Beats - City_Beats_Border_1128-4500".
```

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+4. Set up a lambda or a listener for clicks on the map.
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:17]: Consider rephrasing. We don't need to mention the lambda. Other samples use the following language:

* Connect to the `MouseClicked` event on the `MapView`.
* Set the map to a `MapView` and connect to the `MapQuickView::mouseClicked` signal.
* Listen for `mouseClicked` signal on the map view.
* Create a connection to `MapQuickView::mouseClicked`.

etc

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.metadata.json [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    ],
+    "redirect_from": "",
+    "relevant_apis": [
+        "ArcadeEvaluationResult",
+        "ArcadeEvaluator",
+        "ArcadeExpression",
+        "ArcadeProfile",
+        "Portal",
+        "PortalItem"
+    ],
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.metadata.json:24]: Please add the `relevant_apis` to the `keywords`

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.qml [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.12
+import Esri.ArcGISRuntime 100.15
+import Esri.ArcGISExtras 1.1
+import Esri.ArcGISRuntime.Toolkit 100.14
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/QueryFeaturesWithArcadeExpression.qml:20]: 100.15

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+3. Set the visibility of all the layers to false, except for the layer named "RPD Beats  - City_Beats_Border_1128-4500".
+4. Set up a lambda or a listener for clicks on the map.
+5. Identify the visible layer where it is tapped or clicked on and get the feature.
+6. Create the following `ArcadeExpression`:
+
+    "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n"
+    "return Count(Intersects($feature, crimes));"
+
+7. Create an `ArcadeEvaluator` using the arcade expression and `Enums.ArcadeProfileFormCalculation`.
+8. Create a dictionary of profile variables with the following key-value pairs. This will be passed to `ArcadeEvaluator::evaluate()` in the next step.
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:25]: Function calls should be written in QML style

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+## Use case
+Arcade is a portable, lightweight, and secure expression language used to create custom content in ArcGIS applications. Like other expression languages, it can perform mathematical calculations, manipulate text, and evaluate logical statements. It also supports multi-statement expressions, variables, and flow control statements. What makes Arcade particularly unique when compared to other expression and scripting languages is its inclusion of feature and geometry data types. This sample uses an arcade expression to query the number of crimes in a neighborhood in the last 60 days.
+
+## How to use the sample
+Click on any neighborhood to see the number of crimes in the last 60 days in a callout.
+
+## How it works
+1. Create a `PortalItem` using the URL and ID.
+2. Create a `Map` using the portal item.
+3. Set the visibility of all the layers to false, except for the layer named "RPD Beats  - City_Beats_Border_1128-4500".
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:16]:
```suggestion
3. Set the visibility of all the layers to false, except for the layer named "RPD Beats - City_Beats_Border_1128-4500".
```

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.metadata.json [hunk 12]
```diff
@@ -0,0 +1,29 @@
+{
+    "category": "Display information",
+    "description": "Query features on a map using an arcade expression.",
+    "ignore": false,
+    "images": [
+        "screenshot.png"
+    ],
+    "keywords": [
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.metadata.json:8]: Needs relevant apis

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
+# Query features with arcade expression
+
+Query features on a map using an arcade expression.
+
+![](screenshot.png)
+
+## Use case
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:7]:
```suggestion
## Use case

```
Blank lines should follow headers. Otherwise this sometimes breaks markdown renderers and causes the entire block to be a header.

**Review by tanneryould — CHANGES_REQUESTED** [review]
This looks really good, but there's a few bugs, and a couple stylistic changes that need to be addressed.

Inconsistent behavior between C++ and QML. In C++ `onClicked` toggles the callout -- that is shows it if it's hidden or vice versa. In QML, clicking any feature shows the callout and clicking just the basemap hides the callout. I prefer the QML way of handling this. In C++ if you click the basemap it'll still show the callout and whatever arcade expression was last evaluated.

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md (hunk 6 above)
Comment (Gela) [reply ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:16]: That is the actual name of the layer with 2 spaces

File: ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md (hunk 7 above)
Comment (Gela) [reply ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:17]: Different SDKs handle and verbalize this this differently, for example Android sets up a listener. I will simplify the wording to avoid mentioning lambdas.

File: ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md (hunk 11 above)
Comment (Gela) [reply ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/QueryFeaturesWithArcadeExpression/README.md:16]: same as the other one. That's the actual layer name.
